x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/PasswordAndEnrollmentInitialNode.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.enrollment;
+
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+
+import org.elasticsearch.ExceptionsHelper;
+import org.elasticsearch.cli.EnvironmentAwareCommand;
+import org.elasticsearch.cli.ExitCodes;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.cli.UserException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.settings.SecureSetting;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.xpack.core.XPackSettings;
+import org.elasticsearch.xpack.core.security.user.ElasticUser;
+import org.elasticsearch.xpack.security.tool.CommandLineHttpClient;
+import org.elasticsearch.xpack.security.tool.HttpResponse;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.xpack.security.authc.esnative.tool.SetupPasswordTool.getErrorCause;
+
+public class PasswordAndEnrollmentInitialNode extends EnvironmentAwareCommand {
+    private static final Setting<SecureString> SEED_SETTING = SecureSetting.secureString("keystore.seed", null);
+    private static final Setting<SecureString> BOOTSTRAP_ELASTIC_PASSWORD = SecureSetting.secureString("bootstrap.password",
+        null);
+    private static final Setting<SecureString> CREDENTIALS_PASSWORD = SecureSetting.secureString("bootstrap.password",
+        SEED_SETTING);
```

Comment:
These shouldn't be redefined like this, the existing ones should be referenced here as well.

In addition, what's the plan with CREDENTIALS_PASSWORDS? It's not clear to me.

Reply:
You are right about redefinition, change that.
However, BOOTSTRAP_ELASTIC_PASSWORD in ReservedRealm defined the way it will be backed up by the keystore.seed. I need to look at the bootstrap.password to see if it was provided,  so I have to define a new variable here to read from the keystore.
Credentials password is what BOOTSTRAP_ELASTIC_PASSWORD is in the reserved realm, it will be used to authenticate while sending requests to the node. I created a new variable here to avoid confusion.

Reply:
I think:
```
if (env.settings().hasValue(ReservedRealm.BOOTSTRAP_ELASTIC_PASSWORD.getKey())) {
```
does what you're after, no?